kotlin/workflow-core/src/main/java/com/squareup/workflow/StatefulWorkflow.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow
+
+/**
+ * A composable, stateful object that can [handle events][WorkflowContext.onEvent],
```

Comment:
The docs of the base classes and the interface seem unnecessarily redundant. I don't think a big deal for this PR, but something I'm planning to make a pass on once I'm done with the samples — heads up.

Reply:
Yea, I'm not a fan either. Probably fine to put most of the docs on `Workflow`, point both children to it, and then only have state details on `StatefulWorkflow`.